compiler/circle-input-names/src/circle-input-names.cpp

```diff
@@ -0,0 +1,191 @@
+/*
+ * Copyright (c) 2024 Samsung Electronics Co., Ltd. All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License")
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <iostream>
+#include <crew/PConfigJson.h>
+#include <luci/CircleNodeSummaryBuilders.h>
+
+using namespace luci;
+
+// clang-format off
+// For Variadic Arity Nodes which have multiple inputs.
+template <typename T> struct is_VariadicArity : std::false_type {};
+template <> struct is_VariadicArity<CircleConcatenation> : std::true_type {};
+template <> struct is_VariadicArity<CircleCustom> : std::true_type {};
+template <> struct is_VariadicArity<CirclePack> : std::true_type {};
+template <> struct is_VariadicArity<CircleAddN> : std::true_type {};
+template <> struct is_VariadicArity<CircleIf> : std::true_type {};
+template <> struct is_VariadicArity<CircleWhile> : std::true_type {};
+
+// For Variadic Outputs Nodes which have multiple outputs.
+template <typename T> struct is_VariadicOut : std::false_type {};
+template <> struct is_VariadicOut<CircleCustom> : std::true_type{};
+template <> struct is_VariadicOut<CircleIf> : std::true_type {};
+template <> struct is_VariadicOut<CircleWhile> : std::true_type {};
+// clang-format on
+
+// For Circle Nodes which have variadic arity and variadic outputs
+template <typename CircleOp, typename std::enable_if_t<is_VariadicArity<CircleOp>::value &&
+                                                       is_VariadicOut<CircleOp>::value> * = nullptr>
```

Comment:
Template study really helps me understand this code 👍 
But still I have one question; Why `* = nullptr` is necessary here? 

```suggestion
template <typename CircleOp, typename std::enable_if_t<is_VariadicArity<CircleOp>::value &&
                                                       is_VariadicOut<CircleOp>::value>>
```

Reply:
Ah, I asked ChatGPT and found an answer! 
Because there are several `template<...> auto CircleCreator()` exist, `*=nullptr` looks necessary.